src/views/student-surveys.vue

```diff
@@ -0,0 +1,97 @@
+<template>
+    <div>
+        <h1>students</h1>
+        <div v-if="survey">
+            <survey :survey="survey"></survey>
+        </div>
+        <div v-else>
+            <h3>Loading...</h3>
+        </div>
+    </div>
+</template>
+
+<script>
+import generate_query_string from "./components/generate_query_string";
+import * as SurveyVue from "survey-vue";
+
+var Survey = SurveyVue.Survey;
+export default {
+    components: {
+        Survey
+    },
+    data: function() {
+        return {
+            survey: null,
+            survey_id: null
+        };
+    },
+
+    created: function() {
+        this.getData();
+    },
+
+    methods: {
+        /**
+         * Getting data from API.
+         * All errors are emitted to parent component.
+         */
+        getData: function() {
+            let url = {
+                what: "surveys",
+                user_id: this.$route.params.user_id,
+                survey_id: 1
+            };
+            fetch("get_info.php?" + generate_query_string(url))
+                .then(res => res.json())
+                .then(data => this.parseData(data))
+                .catch(err => {
+                    this.$emit("error", err.toString());
+                });
+        },
+        /**
+         * Function parses questions from API into JSON format and creates
+         * survey.
+         *
+         * @param data Data returned from the API call
+         */
+        parseData: function(data) {
+            if (typeof data == undefined) {
```

Comment:
No `typeof data` is a string. Do `if (data == null)`

Better yet `if (!data || !data.DATA)`